### PR TITLE
Np cristin import persist part of information for patch import

### PR DESCRIPTION
**Pull request by anetteOlli**

### CODE REVIEW

File: cristin-import/src/main/java/no/unit/nva/cristin/lambda/CristinEntryEventConsumer.java [hunk 1]
```diff
@@ -123,6 +127,41 @@ private static boolean isCristinAdditionalIdentifier(AdditionalIdentifier additi
         return CristinObject.IDENTIFIER_ORIGIN.equals(additionalIdentifier.getSource());
     }
 
+    private Publication persistPartOfCristinIdentiferIfPartOfExists(Publication publication,
```
Comment (anetteOlli) [inline cristin-import/src/main/java/no/unit/nva/cristin/lambda/CristinEntryEventConsumer.java:130]: I am not pleased with this code. It is weird. First of all functions that returns an object without modifying it and instead changes stuff in the environment like a void function would. Secondly the multiple creation of cristinObject from the eventbody. 

I am open for suggestion on how to improve this. Right now I can't think of anything without refactor the processInput function to not use attempt, or functions that take in more than two parameters.

Comment (anetteOlli) [inline cristin-import/src/main/java/no/unit/nva/cristin/lambda/CristinEntryEventConsumer.java:130]: I am not pleased with this code. It is weird. First of all functions that returns an object without modifying it and instead changes stuff in the environment like a void function would. Secondly the multiple creation of cristinObject from the eventbody. 

I am open for suggestion on how to improve this. Right now I can't think of anything without refactor the processInput function to not use attempt, or functions that take in more than two parameters.

File: cristin-import/src/main/java/no/unit/nva/cristin/lambda/CristinEntryEventConsumer.java [hunk 2]
```diff
@@ -111,37 +112,88 @@ private static AmazonDynamoDB defaultDynamoDbClient() {
...
+                .withPartOf(
+                    NvaPublicationPartOf.builder()
+                        .withCristinId(publicationIsPartOfThisCristinPublication)
+                        .build())
+                .build();
+    }
+
+    private void persistCristinIdentifierInFileNamedWithPublicationIdentifier(
+        PublicationRepresentations publicationRepresentations) {
+        var cristinIdentifier = publicationRepresentations.getCristinObject().getId().toString();
```
Comment (brinxmat) [inline cristin-import/src/main/java/no/unit/nva/cristin/lambda/CristinEntryEventConsumer.java:168]: I'd be tempted to move this to an accessor method in the PublicationRepresentations object

File: cristin-import/src/main/java/no/unit/nva/cristin/lambda/CristinEntryEventConsumer.java [hunk 3]
```diff
@@ -111,37 +112,88 @@ private static AmazonDynamoDB defaultDynamoDbClient() {
...
-            getCristinIdentifier(publication);
-        var fileUri = constructSuccessFileUri(eventBody, publication);
+    private void persistPartOfCristinIdentifierIfPartOfExists(
+        PublicationRepresentations publicationRepresentations) {
+        if (cristinObjectIsPartOfAnotherPublication(publicationRepresentations.getCristinObject())) {
+            persistPartOfCristinIdentifierWithPublicationId(publicationRepresentations);
+        }
+    }
+
+    private boolean cristinObjectIsPartOfAnotherPublication(CristinObject cristinObject) {
```
Comment (brinxmat) [inline cristin-import/src/main/java/no/unit/nva/cristin/lambda/CristinEntryEventConsumer.java:135]: I'd be tempted to move this to the PublicationRepresentations class

File: cristin-import/src/main/java/no/unit/nva/cristin/lambda/CristinEntryEventConsumer.java [hunk 4]
```diff
@@ -111,37 +112,88 @@ private static AmazonDynamoDB defaultDynamoDbClient() {
...
+    private void persistPartOfCristinIdentifierWithPublicationId(
+        PublicationRepresentations publicationRepresentations) {
+        var partOf = createPartOf(publicationRepresentations);
+        var fileUri = constructPartOfFileUri(publicationRepresentations);
+        var s3Driver = new S3Driver(s3Client, fileUri.getUri().getHost());
+        attempt(() -> s3Driver.insertFile(fileUri.toS3bucketPath(), partOf.toJsonString())).orElseThrow();
+    }
+
+    private NvaPublicationPartOfCristinPublication createPartOf(PublicationRepresentations publicationRepresentations) {
+        var publicationIdentifier = publicationRepresentations.getPublication().getIdentifier().toString();
```
Comment (brinxmat) [inline cristin-import/src/main/java/no/unit/nva/cristin/lambda/CristinEntryEventConsumer.java:153]: I believe in general, the internals of the access to values in the contents of the PublicationRepresentations should be hidden — for the methods here, they should not need to know the internal structures as it makes the code more intertwined than need be.

File: cristin-import/src/main/java/no/unit/nva/cristin/lambda/CristinEntryEventConsumer.java [hunk 5]
```diff
@@ -111,37 +112,88 @@ private static AmazonDynamoDB defaultDynamoDbClient() {
...
+    private Publication persistConversionReports(PublicationRepresentations publicationRepresentations) {
+        persistCristinIdentifierInFileNamedWithPublicationIdentifier(publicationRepresentations);
+        persistPartOfCristinIdentifierIfPartOfExists(publicationRepresentations);
+        return publicationRepresentations.getPublication();
     }
 
-    private static boolean isCristinAdditionalIdentifier(AdditionalIdentifier additionalIdentifier) {
-        return CristinObject.IDENTIFIER_ORIGIN.equals(additionalIdentifier.getSource());
+    private PublicationRepresentations mapEventToNvaPublication(
+        FileContentsEvent<JsonNode> eventbody) {
```
[the remark this answers is not in the document]
Comment (anetteOlli) [reply cristin-import/src/main/java/no/unit/nva/cristin/lambda/CristinEntryEventConsumer.java]: fikset

File: cristin-import/src/main/java/no/unit/nva/cristin/lambda/CristinEntryEventConsumer.java (hunk 2 above)
Comment (anetteOlli) [reply cristin-import/src/main/java/no/unit/nva/cristin/lambda/CristinEntryEventConsumer.java]: Fikset

File: cristin-import/src/main/java/no/unit/nva/cristin/lambda/CristinEntryEventConsumer.java (hunk 3 above)
Comment (anetteOlli) [reply cristin-import/src/main/java/no/unit/nva/cristin/lambda/CristinEntryEventConsumer.java]: fikset

File: cristin-import/src/main/java/no/unit/nva/cristin/lambda/CristinEntryEventConsumer.java (hunk 4 above)
Comment (anetteOlli) [reply cristin-import/src/main/java/no/unit/nva/cristin/lambda/CristinEntryEventConsumer.java]: fikset

File: cristin-import/src/main/java/no/unit/nva/cristin/lambda/CristinEntryEventConsumer.java [hunk 6]
```diff
@@ -111,37 +112,88 @@ private static AmazonDynamoDB defaultDynamoDbClient() {
...
-                   .map(AdditionalIdentifier::getValue);
+    private Publication persistConversionReports(PublicationRepresentations publicationRepresentations) {
+        persistCristinIdentifierInFileNamedWithPublicationIdentifier(publicationRepresentations);
+        persistPartOfCristinIdentifierIfPartOfExists(publicationRepresentations);
+        return publicationRepresentations.getPublication();
     }
 
-    private static boolean isCristinAdditionalIdentifier(AdditionalIdentifier additionalIdentifier) {
-        return CristinObject.IDENTIFIER_ORIGIN.equals(additionalIdentifier.getSource());
+    private PublicationRepresentations mapEventToNvaPublication(
```
[the remark this answers is not in the document]
Comment (anetteOlli) [reply cristin-import/src/main/java/no/unit/nva/cristin/lambda/CristinEntryEventConsumer.java]: fikset

**Review by brinxmat — APPROVED** [review]

**Pull request merged**